akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala

```diff
@@ -607,17 +607,24 @@ class TcpSpec extends StreamSpec("akka.stream.materializer.subscription-timeout.
       // make sure server has started
       serverBound.futureValue
 
-      val connectAndCountBytes = Source(immutable.Iterable.fill(100)(ByteString(0)))
+      val firstProbe = TestPublisher.probe[ByteString]()
+      val firstResult = Source.fromPublisher(firstProbe)
         .via(Tcp().outgoingConnection(address))
-        .fold(0)(_ + _.size).toMat(Sink.head)(Keep.right)
-
-      val total = connectAndCountBytes.run()
+        .runWith(Sink.seq)
 
+      // wait until the first client has connected
       awaitAssert(firstClientConnected.future, 2.seconds)
+      firstProbe.expectRequest()
+
+      // then connect the second one, which will be ignored
+      val secondProbe = TestPublisher.probe[ByteString]()
+      val rejected = Source.fromPublisher(secondProbe).via(Tcp().outgoingConnection(address)).runWith(Sink.seq)
 
-      val rejected = connectAndCountBytes.run()
-      total.futureValue(Timeout(10.seconds)) should ===(100)
```

Comment:
`total` probably failed because once `grouped(2).take(1)` got two input elements the `bind` flow would have been cancelled, bringing down the associated connections.

Not sure why it is stable now, though ;)